kidex-client/src/main.rs

```diff
@@ -14,31 +15,168 @@ enum Command {
     Shutdown,
     ReloadConfig,
     RegenerateIndex,
+    /// Return the entire index
     GetIndex { path: Option<PathBuf> },
+    /// Get the index and filters the results
+    Find {
+        // TODO: Add some CLI arguments:
+        // --root <path>
+        // --mode <mode> | --regex | --literal | --smart (default)
+
+        /// Case-insensitive search (default: smart-case)
+        #[arg(short, long, group = "case")]
```

Comment:
I think a lot of these arguments should be turned into Query arguments as outlined in another comment.